### ISSUE TITLE
[BUG] web: field context is not passed to the record context

### ISSUE THREAD

**Issue by blaggacao** [odoo/odoo#29203]
Note: I think this one is for @ged-odoo. It assume in-depth knowledge of the web refactoring.

A field context `{'user_preference': True}` defined like this
```python

class Users(models.Model):
    _inherit = ['res.users']
    reporting_standard_ids = fields.Many2many('account.reporting.standard', string='Available Reporting Standards', context={'user_preference': True})
```
or (equivalently) like this
```xml

<field name="reporting_standard_ids" widget="many2many_tags" options="{'no_create': True}" context="{'user_preference': True}"/>
```
**is not passed down to the record context in the js model**, therefore

```js
    _applyChange: function (recordID, changes, options) {
        var self = this;
        var record = this.localData[recordID];
        var field;
        var defs = [];
        options = options || {};
        record._changes = record._changes || {};
        if (!options.doNotSetDirty) {
            record._isDirty = true;
        }
        var initialData = {};
        this._visitChildren(record, function (elem) {
            initialData[elem.id] = $.extend(true, {}, _.pick(elem, 'data', '_changes'));
        });

        // apply changes to local data
        for (var fieldName in changes) {
            field = record.fields[fieldName];
            if (field.type === 'one2many' || field.type === 'many2many') {
                defs.push(this._applyX2ManyChange(record, fieldName, changes[fieldName], options.viewType, options.allowWarning));
            } else if (field.type === 'many2one' || field.type === 'reference') {
                defs.push(this._applyX2OneChange(record, fieldName, changes[fieldName]));
            } else {
                record._changes[fieldName] = changes[fieldName];
            }
        }
   ...
```
fails. Specifically the `rpc read` in
```js
_applyX2ManyChange
```

when there exists a security rule
```xml
  <record id="account_mickey_mouse_rule" model="ir.rule">
      <field name="name">Mickey Mouse Reporting Standard</field>
      <field name="model_id" ref="model_account_reporting_standard"/>
      <field name="global" eval="True"/>
      <field name="domain_force">[('id','in',user.reporting_standard_ids.ids)]</field>
  </record>
```
and the `read` method has been specifically prepared like so:
```python
    @api.multi
    def read(self, fields=None, load='_classic_read'):
        context = dict(self.env.context)
        newself = self
        if context.get('user_preference', None):
            newself = newself.sudo()
        return super(AccountReportingStandard, newself.with_context(context)).read(fields=fields, load=load)
```

**Comment by blaggacao** [comment]
@ged-odoo As this is specific to the web client, is my description useful to you? Maybe this api usage was an oversight in the web client refactoring... Or, maybe, is the context key not supported any more?

This should be fixed in v12 and possibly v11?

**Comment by blaggacao** [comment]
Ticket
#1914466
submitted

**Comment by ged-odoo** [comment]
@blaggacao The description looks fine. This will be handled by the bug team I assume :)

**Comment by nle-odoo** [comment]
This should be solved in 11.0 with https://github.com/odoo/odoo/commit/3125d20d
